### PR TITLE
camera implementation

### PR DESCRIPTION
**Pull request by shivasku82**

### CODE REVIEW

File: camera/MultiCameraApplication/java/com/intel/multicamera/CameraBase.java [hunk 1]
```diff
@@ -124,247 +121,138 @@
...
-        if (recording) {
-            mRecordingTimeView.setText("");
-            mRecordingTimeView.setVisibility(View.VISIBLE);
-            mRecordingTimeView.announceForAccessibility(
-                    mActivity.getResources().getString(R.string.video_recording_started));
+                MultiViewActivity Mactivity = (MultiViewActivity) mActivity;
+                Mactivity.closeCamera();
+                ic_camera.setIsCameraOrSurveillance(0);
+                Intent intent = new Intent(mActivity, SingleCameraActivity.class);
+                intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK|Intent.FLAG_ACTIVITY_CLEAR_TOP);
```
Comment (JeevakaPrabu) [inline camera/MultiCameraApplication/java/com/intel/multicamera/CameraBase.java:185]: make sure number of characters/line is below 100

File: camera/MultiCameraApplication/java/com/intel/multicamera/CameraBase.java [hunk 2]
```diff
@@ -72,40 +77,32 @@
...
+    private CameraBase mCameraBase;
     private static final String SIZE_HD = "HD 720p";
 
     /**
      * Whether the app is recording video now
      */
     private boolean mIsRecordingVideo;
-
+    private  MultiCamera ic_camera;
     // The video file that the hardware camera is about to record into
```
Comment (JeevakaPrabu) [inline camera/MultiCameraApplication/java/com/intel/multicamera/CameraBase.java:101]: Comments in line no: 101 and 102 are not relevant and needs to be removed

File: camera/MultiCameraApplication/java/com/intel/multicamera/FullScreenActivity.java [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+                getFragmentManager().beginTransaction().remove(Fragment).commit();
+
+                v.setVisibility(v.GONE);
+                mSettings.setVisibility(View.VISIBLE);
+
+                mCameraRecord.setVisibility(View.VISIBLE);
+                mCameraPicture.setVisibility(View.VISIBLE);
+                mCameraSwitch.setVisibility(View.VISIBLE);
+                mCameraSplit.setVisibility(View.VISIBLE);
+                //ic_camera.getmTopLeftCam().createCameraPreview();
```
Comment (JeevakaPrabu) [inline camera/MultiCameraApplication/java/com/intel/multicamera/FullScreenActivity.java:175]: remove commented out code

File: camera/MultiCameraApplication/java/com/intel/multicamera/FullScreenActivity.java [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+            }
+        });
+
+        mCameraSplit.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                MultiCamera ic_camera = MultiCamera.getInstance();
+                ic_camera.setIsCameraOrSurveillance(1);
+                Intent intent = new Intent(FullScreenActivity.this, MultiViewActivity.class);
+                intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK|Intent.FLAG_ACTIVITY_CLEAR_TOP);
```
Comment (JeevakaPrabu) [inline camera/MultiCameraApplication/java/com/intel/multicamera/FullScreenActivity.java:186]: make sure number of characters/line is < 100

File: camera/MultiCameraApplication/java/com/intel/multicamera/SingleCameraActivity.java [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+            actionBar.setHomeButtonEnabled(true);
+        }
+
+        setContentView(R.layout.activity_full_screen);
+
+        mIsRecordingVideo = false;
+        mCameraInst = MultiCamera.getInstance();
+        mCameraSwitch = findViewById(R.id.camera_switch);
+        mCameraSwitch.setVisibility(View.VISIBLE);
+        if(mCameraInst.getOpenCameraId() == 2 || mCameraInst.getOpenCameraId() == 3)
```
[the remark this answers is not in the document]
Comment (shivasku82) [reply camera/MultiCameraApplication/java/com/intel/multicamera/SingleCameraActivity.java]: camera 0 is back 1 is front with 4 camera connected if user click on camera 2 or 3 then camera switch functionality wont be seen

File: camera/MultiCameraApplication/java/com/intel/multicamera/SingleCameraActivity.java [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+        super.onPause();
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        Log.e(TAG, " onResume");
+        /*MultiCamera ic_cam = MultiCamera.getInstance();
+        ic_cam.setIsCameraOrSurveillance(0);
+        OpenCamera();*/
```
[the remark this answers is not in the document]
Comment (shivasku82) [reply camera/MultiCameraApplication/java/com/intel/multicamera/SingleCameraActivity.java]: activity always in top clear hence on activity oncreate will be called there registration happen however complete hotplug feature not tested for all scenario's

File: camera/MultiCameraApplication/java/com/intel/multicamera/FullScreenActivity.java [hunk 7]
```diff
@@ -0,0 +1,434 @@
...
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        Log.e(TAG, "onResume");
+        MultiCamera ic_cam = MultiCamera.getInstance();
+        ic_cam.setIsCameraOrSurveillance(0);
+        if (ic_cam.getWhichCamera() == 0) {
+            openBackCamera();
+        }
```
[the remark this answers is not in the document]
Comment (shivasku82) [reply camera/MultiCameraApplication/java/com/intel/multicamera/FullScreenActivity.java:446]: can you please paste changes here

File: camera/MultiCameraApplication/java/com/intel/multicamera/CameraBase.java [hunk 8]
```diff
@@ -62,8 +64,11 @@
 import java.util.Arrays;
 import java.util.List;
 import java.util.Optional;
+import java.util.concurrent.TimeUnit;
 
-public class CameraBase implements MediaRecorder.OnErrorListener, MediaRecorder.OnInfoListener {
+import javax.crypto.Mac;
```
[the remark this answers is not in the document]
Comment (shivasku82) [reply camera/MultiCameraApplication/java/com/intel/multicamera/CameraBase.java:69]: not required removed

**Pull request closed without merging**